src/Helpers/NotificationHelper.php

```diff
@@ -353,13 +353,20 @@ public function getNotificationMessage(string $category, string $type, array $da
                         throw new \LogicException("Repelled spy op notification for operation key {$data['operationKey']} not yet implemented");
                 }
 
+                $lossString = "";
+                foreach ($data['spiesKilled'] as $key => $value) {
+                    $lossString .= number_format($value) . " of our " . str_plural($key, $value);
```

Comment:
I don't think " of our " would make sense here, seeing the context.

I.e. "'Spies from Foo (#1) were discovered in our vaults! We executed 12 of our spies.',

Should probably be "of their".

In addition, the helper function `generate_sentence_from_array()` exists in `src/helpers.php`. Please use that for this